code/game/objects/items/devices/radio/encryptionkey.dm

```diff
@@ -141,13 +141,51 @@
 	name = "\improper ERT radio encryption key"
 	channels = list(CHANNEL_RESPONSE_TEAM = TRUE)
 
-/obj/item/device/encryptionkey/entertainment
-	name = "entertainment radio key"
-	channels = list(CHANNEL_ENTERTAINMENT = TRUE)
-
 /obj/item/device/encryptionkey/rev
 	name = "standard encryption key"
 	desc = "An encryption key for a radio headset. Contains cypherkeys."
-	channels = list(CHANNEL_COMMON = TRUE, CHANNEL_ENTERTAINMENT = TRUE, CHANNEL_RAIDER = TRUE)
+	additional_channels = list(CHANNEL_RAIDER = TRUE)
 	origin_tech = list(TECH_ILLEGAL = 2)
 	desc_antag = "An encryption key that allows you to speak on private non-station channels. Use :x to access the private channel."
+
+/obj/item/device/encryptionkey/eng_spare
+	name = "spare engineering radio encryption key"
+	additional_channels = list(CHANNEL_ENGINEERING = TRUE)
+
+/obj/item/device/encryptionkey/med_spare
+	name = "spare medical radio encryption key"
+	additional_channels = list(CHANNEL_MEDICAL = TRUE)
+
+/obj/item/device/encryptionkey/sec_spare
+	name = "spare security radio encryption key"
+	additional_channels = list(CHANNEL_SECURITY = TRUE)
+
+/obj/item/device/encryptionkey/sci_spare
+	name = "spare science radio encryption key"
+	additional_channels = list(CHANNEL_SCIENCE = TRUE)
+
+
+// Encryption Key Pouch
+/obj/item/storage/box/fancy/keypouch
+	name = "encryption key pouch"
+	desc = "A pouch designed to store three encryption keys."
+	icon = 'icons/obj/radio.dmi'
+	icon_state = "keypouch3" // make this keypouch0 if you want to map in an empty one
```

Comment:
Could it maybe update its icon based on remaining encryption keys? And in Initialize?